src/panel/service_panel.js

```diff
@@ -1,47 +1,81 @@
-import ServicePanelView from '../views/service_panel.dot';
-import Panel from '../libs/panel';
-import nconf from '@qwant/nconf-getter';
+/* globals _ */
+import React from 'react';
+import ReactDOM from 'react-dom';
 import CategoryService from '../adapters/category_service';
 
-export default class ServicePanel {
-  constructor() {
-    this.panel = new Panel(this, ServicePanelView);
-    this.categories = CategoryService.getCategories();
-    this.mustDeployCategories = this.categories.length > 8;
-    this.isDeployed = false;
-    this.isDirectionActive = nconf.get().direction.enabled;
-    this.active = true;
-  }
+// Returns a function, which when called will navigate to the url
+const navTo = url => () => {
+  window.app.navigateTo(url);
+};
 
-  toggleCategories() {
-    this.isDeployed = !this.isDeployed;
-    this.panel.update();
-  }
+const CategoryButton = ({ category }) => {
+  return <button
+    className="service_panel__category"
+    type="button"
+    onClick={navTo(`/places/?type=${category.name}`) }
+  >
+    <div className="service_panel__category__icon" style={{
+      background: category.backgroundColor,
+    }}>
+      <span className={`icon icon-${category.iconName}`} />
+    </div>
+    <div className="service_panel__category__title">{category.label}</div>
+  </button>;
+};
+
+class ServicePanel extends React.Component {
+  render() {
+    const categories = this.props.categories;
+
+    return <div className="service_panel">
+      <h3 className="service_panel__title">{_('Qwant Maps services', 'service panel')}</h3>
+
+      {/* TODO: use real links instead of all these buttons */}
 
-  openFavorite() {
-    window.app.navigateTo('/favs');
+      <div className="service_panel__actions">
+        <button
+          className="service_panel__action service_panel__item__fav"
+          onClick={navTo('/favs') }
+        >
+          <i className="icon-icon_star" />{' '}
```

Comment:
In JSX, empty tags can use the XML shortcut closing syntax `<foo />`. React knows which HTML elements it has to really render as opening/closing tags.